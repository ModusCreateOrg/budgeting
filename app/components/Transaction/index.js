import React from 'react';
import PropTypes from 'prop-types';

const TransactionComponent = props => {
  console.log(props);
  return (
    <div>
      <button
        onClick={() => {
          props.backToPreviousRoute();
          console.log('onclick');
        }}
      >
        Back
      </button>
      <div>
        <h2>
          {props.transaction.category} - {props.transaction.description}
        </h2>
      </div>
    </div>
  );
};
TransactionComponent.propTypes = {
  transaction: PropTypes.object.isRequired,
  backToPreviousRoute: PropTypes.func.isRequired,
};

TransactionComponent.defaultProps = {
  transaction: {},
};

export default TransactionComponent;
