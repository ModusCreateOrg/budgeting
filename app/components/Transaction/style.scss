@import 'theme/variables';

.title {
  margin-bottom: 0;
}

.negative {
  color: $red;
}

.positive {
  color: $green;
}
