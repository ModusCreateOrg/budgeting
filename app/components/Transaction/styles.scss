@import 'theme/variables';

.neg {
  color: $red;
  margin-left: 10px;
}

.pos {
  color: $green;
}

.content {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
.subtitle {
  display: flex;
  flex-direction: row;
  margin-bottom: 5%;
}
