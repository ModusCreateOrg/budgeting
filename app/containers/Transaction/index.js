import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { injectAsyncReducers } from 'store';
import transactionReducer from 'modules/transactions';
import categoryReducer from 'modules/categories';

import { getTransactionById } from 'selectors/transactions';
import NotFound from 'components/NotFound';
import TransactionDetails from 'containers/TransactionDetails';

injectAsyncReducers({
  transactions: transactionReducer,
  categories: categoryReducer,
});

@connect((state, { transactionId }) => ({
  transaction: getTransactionById(transactionId)(state),
}))
export default class Transaction extends PureComponent {
  static propTypes = {
    transactionId: PropTypes.string.isRequired,
  };

  render() {
    const { transactionId, transaction } = this.props;

    return (
      <div>
        <Link to="/budget">‚Üê Back</Link>
        {transaction ? (
          <TransactionDetails transaction={transaction} />
        ) : (
          <NotFound title="Transaction Not Found" description={`Transaction with ID "${transactionId}" not found.`} />
        )}
      </div>
    );
  }
}
