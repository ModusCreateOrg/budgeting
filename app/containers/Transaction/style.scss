@import 'theme/variables';

.transaction {
  display: flex;
  flex-direction: column;
  button {
    box-shadow: 1px 1px 2px 1px black;
    padding: 10px;
    text-align: center;
    margin-bottom: 10px;
  }
}
