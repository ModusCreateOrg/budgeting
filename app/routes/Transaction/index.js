import React, { PureComponent } from 'react';
import Chunk from 'components/Chunk';

const loadTransactionContainer = () => import('containers/Transaction' /* webpackChunkName: "budget-item" */);

export default class Transaction extends PureComponent {
  render() {
    const { match } = this.props;
    return <Chunk load={loadTransactionContainer} transactionId={match.params.transactionId} />;
  }
}
