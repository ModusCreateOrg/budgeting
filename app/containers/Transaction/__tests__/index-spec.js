import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router';
import Transaction from '../';

jest.mock('containers/TransactionDetails');

it('renders with transaction', () => {
  const tree = renderer
    .create(
      <StaticRouter location="transaction" context={{}}>
        <Transaction.WrappedComponent
          transaction={{
            id: 1,
          }}
          transactionId="1"
        />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders not found', () => {
  const tree = renderer
    .create(
      <StaticRouter location="transaction" context={{}}>
        <Transaction.WrappedComponent transactionId="5" />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
