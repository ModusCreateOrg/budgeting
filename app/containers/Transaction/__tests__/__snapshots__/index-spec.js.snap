// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders not found 1`] = `
<div>
  <a
    href="/budget"
    onClick={[Function]}
  >
    ← Back
  </a>
  <div>
    <h1>
      Transaction Not Found
    </h1>
    <p>
      Transaction with ID "5" not found.
    </p>
  </div>
</div>
`;

exports[`renders with transaction 1`] = `
<div>
  <a
    href="/budget"
    onClick={[Function]}
  >
    ← Back
  </a>
</div>
`;
